BUILD

```diff
@@ -0,0 +1,248 @@
+    srcs = glob([
+        "src/api/*.cc",
+        "src/api/rules/*.cc",
+    ]),
+    hdrs = glob([
+        "include/maliput/api/*.h",
+        "include/maliput/api/rules/*.h",
+        "include/maliput/base/*.h",
+        "include/maliput/common/*.h",
+        "include/maliput/math/*.h",
+    ]),
+    strip_include_prefix = "include",
+)
+
+cc_library(
+    name = "base",
+    srcs = glob(["src/base/*.cc"]),
+    hdrs = glob([
+        "include/maliput/base/*.h",
+    ]),
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+        "@yaml-cpp//:yaml-cpp",
+    ],
+)
+
+cc_library(
+    name = "common",
+    srcs = glob(["src/common/*.cc"]),
+    hdrs = glob(["include/maliput/common/*.h"]),
+    strip_include_prefix = "include",
+    deps = ["@fmt"],
+)
+
+cc_library(
+    name = "geometry_base",
+    srcs = glob(["src/geometry_base/*.cc"]),
+    hdrs = glob([
+        "include/maliput/geometry_base/*.h",
+    ]),
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+    ],
+)
+
+cc_library(
+    name = "math",
+    srcs = glob(["src/math/*.cc"]),
+    hdrs = glob([
+        "include/maliput/math/*.h",
+    ]),
+    strip_include_prefix = "include",
+    deps = ["@//:common"],
+)
+
+cc_library(
+    name = "plugin",
+    srcs = glob(["src/plugin/*.cc"]),
+    hdrs = glob([
+        "include/maliput/plugin/*.h",
+        "include/maliput/api/*.h",
+        "include/maliput/math/*.h",
+    ]),
+    linkopts = ["-ldl"],
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+        "@//:utility",
+    ],
+)
+
+cc_library(
+    name = "routing",
+    srcs = glob(["src/routing/*.cc"]),
+    hdrs = glob([
+        "include/maliput/routing/*.h",
+    ]),
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+    ],
+)
+
+cc_library(
+    name = "test_utilities",
+    srcs = glob(["src/test_utilities/*.cc"]),
+    hdrs = glob([
+        "include/maliput/test_utilities/*.h",
+    ]),
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:geometry_base",
+        "@//:math",
+        "@com_google_googletest//:gtest",
+    ],
+)
+
+cc_library(
+    name = "utility",
+    srcs = glob(["src/utility/*.cc"]),
+    hdrs = glob(["include/maliput/utility/*.h"]),
+    linkopts = ["-lstdc++fs"],
+    strip_include_prefix = "include",
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+    ],
+)
+
+#### test #####
+cc_test(
+    name = "api_test",
+    size = "enormous",
+    timeout = "long",
+    srcs = glob(["test/api/*.cc"]),
+    deps = [
+        "@//:api",
+        "@//:common",
+        "@//:math",
+        "@//:test_utilities",
+        "@//:utility",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
```

Comment:
@erasmomontes meta: if I may suggest, we could have one test per file as it is done in CMake. 

You can do that by combining the `glob` function with list comprehensions ie.:

```python

def basename(path):
    _, _, name = path.rpartition("/")
    _, _, basename = name.rpartition(".")
    return basename

[cc_test(
     name = "api_" + basename(src_file), 
     srcs = [src_file], # and the rest...
 ) for src_file in glob(["test/api/*.cc"])]
```